subprojects/code-quality/src/testFixtures/groovy/org/gradle/quality/integtest/fixtures/CodeNarcCoverage.groovy

```diff
@@ -21,21 +21,30 @@ import org.gradle.api.plugins.quality.CodeNarcPlugin
 import org.gradle.util.internal.VersionNumber
 
 class CodeNarcCoverage {
+    public static final List<String> ALL = [CodeNarcPlugin.DEFAULT_CODENARC_VERSION, "0.17", "0.21", "0.23", "0.24.1", "0.25.2", "1.0", "1.6.1", "2.0.0", "2.2.0", "3.0.1"].asImmutable()
```

Comment:
This is not your problem to deal with and I don't want to hold up this PR, but how do we determine which versions we test here? I'd imagine its just "I want to test a new version so let me add that version to the list", without consideration for removing old versions. 

`0.17` came out in 2012, `0.21` in 2014, `0.23` and `0.24.1` in 2015, and `0.25.2` in 2016. All of these are before even an explicit `1.0` release`. How far back should we claim support for?

Reply:
That's a great question, and I suspect you're right that we have just accumulated these versions over time.  It would be nice to have some sort of well understood strategy for these 3rd party tools, such as "we only support/test the last two major versions" or something like that.  Which would not necessarily mean that it might not work with older versions, only that we don't test or advertise support for those older versions.